frontend/src/citizen-frontend/calendar/HolidayModal.tsx

```diff
@@ -2,17 +2,90 @@
 //
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
-import React from 'react'
+import React, { useState, useCallback } from 'react'
+import styled from 'styled-components'
 
 import { renderResult } from 'citizen-frontend/async-rendering'
 import { useLang, useTranslation } from 'citizen-frontend/localization'
-import { ReservationChild } from 'lib-common/generated/api-types/reservations'
+import { Result } from 'lib-common/api'
+import FiniteDateRange from 'lib-common/finite-date-range'
+import { FreeAbsencePeriod } from 'lib-common/generated/api-types/holidayperiod'
+import {
+  AbsenceRequest,
+  ReservationChild
+} from 'lib-common/generated/api-types/reservations'
 import ExternalLink from 'lib-components/atoms/ExternalLink'
+import Select from 'lib-components/atoms/dropdowns/Select'
+import { FixedSpaceColumn } from 'lib-components/layout/flex-helpers'
 import { AsyncFormModal } from 'lib-components/molecules/modals/FormModal'
+import { H2, Label } from 'lib-components/typography'
+import { defaultMargins } from 'lib-components/white-space'
 
 import { useHolidayPeriods } from '../holiday-periods/state'
 
-interface Props {
+import { postAbsences } from './api'
+
+const formatChildName = (child: ReservationChild) =>
+  child.preferredName || child.firstName.split(' ')[0]
+
+type HolidaySelectorProps = {
+  child: ReservationChild
+  freeAbsencePeriod: FreeAbsencePeriod | null
+  value: FiniteDateRange | null
+  onSelectPeriod: (selection: FiniteDateRange | null) => void
+}
+
+const HolidaySelector = React.memo(function HolidaySelector({
+  child,
+  freeAbsencePeriod,
+  value,
+  onSelectPeriod
+}: HolidaySelectorProps) {
+  const i18n = useTranslation()
+  const [lang] = useLang()
+
+  type HolidayOption = {
+    name: string
+    period: FiniteDateRange | null
+  }
+  const options: HolidayOption[] = [
+    {
+      name: '',
+      period: null
+    },
+    ...(freeAbsencePeriod
+      ? freeAbsencePeriod.periodOptions.map((period) => ({
+          name: period.format(),
+          period
+        }))
+      : [])
+  ]
+
+  return (
+    <div>
+      <H2>
+        {i18n.calendar.holidayModal.holidayFor} {formatChildName(child)}
+      </H2>
+      {freeAbsencePeriod && (
+        <>
+          <Label>{freeAbsencePeriod.periodOptionLabel[lang]}</Label>
+          <Select
+            items={options}
+            selectedItem={
+              options.find(({ period }) => period == value) ?? options[0]
```

Comment:
`options[0]` is a little bit ambiguous. Extracting the empty option to a constant and using it here would make it more transparent.